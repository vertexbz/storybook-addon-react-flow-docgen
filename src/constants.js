// @flow
const PluginId = 'storybook/react-flow-docgen';

export const id = {
    Plugin: PluginId,
    Panel: PluginId + '/panel'
};

export const event = {
    LoadDocData: PluginId + '/loadDocData'
};
