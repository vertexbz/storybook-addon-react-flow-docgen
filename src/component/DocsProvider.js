// @flow
import * as React from 'react';
import addons from '@storybook/addons';
import { event } from '../constants';
import type { DocgenInfoType } from '../types';

type DocsProviderProps<E> = {
    docs?: Array<DocgenInfoType>,
    children: React.Element<E>
};

export default
class DocsProvider<E: *> extends React.Component<DocsProviderProps<E>> {
    componentDidMount() {
        addons.getChannel().emit(event.LoadDocData, this.props.docs || []);
    }

    componentWillUnmount() {
        addons.getChannel().emit(event.LoadDocData, []);
    }

    render(): React.Element<E> {
        return this.props.children;
    }
}
