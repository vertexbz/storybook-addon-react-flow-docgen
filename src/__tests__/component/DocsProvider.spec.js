import * as React from 'react';
import { mount } from 'enzyme';
import DocsProvider from '../../component/DocsProvider';
import { event } from '../../constants';

jest.mock('@storybook/addons');
const addonsMock = require('@storybook/addons').default;

describe('DocsProvider', () => {
    it('should work properly', () => {
        const docs = [
            {
                fakeDocsOne: 'docs 1'
            },
            {
                fakeDocsTwo: 'docs 2'
            }
        ];

        const TestStory = () => <div />;

        const wrapper = mount(
            <DocsProvider docs={docs}>
                <TestStory />
            </DocsProvider>
        );


        expect(addonsMock.__channel.emit).toBeCalledTimes(1);
        expect(addonsMock.__channel.emit).toBeCalledWith(event.LoadDocData, docs);
        addonsMock.__channel.emit.mockClear();

        expect(wrapper.render()).toMatchSnapshot();

        wrapper.unmount();
        expect(addonsMock.__channel.emit).toBeCalledTimes(1);
        expect(addonsMock.__channel.emit).toBeCalledWith(event.LoadDocData, []);
    });
});
