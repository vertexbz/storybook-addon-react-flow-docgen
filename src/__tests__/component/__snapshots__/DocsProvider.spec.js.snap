// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocsProvider should work properly 1`] = `<div />`;
